File: branches/Medusa/Source/Fantasy.Jobs/Solar/SatelliteHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ServiceModel;
using Fantasy.Jobs.Management;
using Fantasy.ServiceModel;

namespace Fantasy.Jobs.Solar
{
    [ServiceBehavior(ConcurrencyMode=ConcurrencyMode.Reentrant, InstanceContextMode=InstanceContextMode.PerSession, Namespace=Consts.JobServiceNamespaceURI)]
    public class SatelliteHandler : ISatelliteHandler
    {
        public SatelliteHandler()
        {

        }

        private ISatellite _satellite;
        private string _name;
        #region ISatelliteHandler Members

        public void Connect(string name)
        {
            _name = name;
            _satellite = OperationContext.Current.GetCallbackChannel<ISatellite>();
            SatelliteManager manager = JobManager.Default.GetRequiredService<SatelliteManager>();
            manager.RegisterSatellite(name, _satellite);
        }

        

        public void Echo()
        {
            SatelliteManager manager = JobManager.Default.GetRequiredService<SatelliteManager>();
            if (!manager.IsValid(_satellite))
            {
                throw new FaultException<CallbackExpiredException>(new CallbackExpiredException(), new FaultReason("Satellite handle is invalid."));
            }
            else
            {
                manager.UpdateEchoTime(_name);
            }
        }

        public void Disconnect()
        {
            SatelliteManager manager = JobManager.Default.GetRequiredService<SatelliteManager>();
            manager.UnregisterSatellite(_satellite);
        }

        #endregion

     
    }
}

File: AutoConverted/Fantasy.Jobs/Solar/SatelliteHandler.java
package Fantasy.Jobs.Solar;

import Fantasy.Jobs.Management.*;
import Fantasy.ServiceModel.*;

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
//[ServiceBehavior(ConcurrencyMode=ConcurrencyMode.Reentrant, InstanceContextMode=InstanceContextMode.PerSession, Namespace=Consts.JobServiceNamespaceURI)]
public class SatelliteHandler implements ISatelliteHandler
{
	public SatelliteHandler()
	{

	}

	private ISatellite _satellite;
	private String _name;
//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#region ISatelliteHandler Members

	public final void Connect(String name)
	{
		_name = name;
		_satellite = OperationContext.Current.<ISatellite>GetCallbackChannel();
		SatelliteManager manager = JobManager.getDefault().<SatelliteManager>GetRequiredService();
		manager.RegisterSatellite(name, _satellite);
	}



	public final void Echo()
	{
		SatelliteManager manager = JobManager.getDefault().<SatelliteManager>GetRequiredService();
		if (!manager.IsValid(_satellite))
		{
			throw new FaultException<CallbackExpiredException>(new CallbackExpiredException(), new FaultReason("Satellite handle is invalid."));
		}
		else
		{
			manager.UpdateEchoTime(_name);
		}
	}

	public final void Disconnect()
	{
		SatelliteManager manager = JobManager.getDefault().<SatelliteManager>GetRequiredService();
		manager.UnregisterSatellite(_satellite);
	}

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#endregion


}